DorisProject/src/app/pages/stock/stock.component.ts

```diff
@@ -66,8 +70,34 @@ export class StockComponent implements OnInit {
   }
 
   addProduct() {
-    this.router.navigate(['add-stock']);
+    this.router.navigate(['add-stock'], {state: {data: null}});
```

Comment:
Codacy has a fix for the issue: [Replace `state:·{data:·null}` with `·state:·{·data:·null·}·`](https://app.codacy.com/gh/SaraArenasV/DorisProjectFrontEnd/pullRequest?prid=9269670)
```suggestion
    this.router.navigate(['add-stock'], { state: { data: null } });
```